### ISSUE TITLE
"Got fatal error on delete" error message should include path

### ISSUE THREAD

**Issue by folknor** [rclone/rclone#7355]
#### What is your current rclone version (output from `rclone version`)?
1.64

#### What problem are you are trying to solve?
The console output from an `rclone` command like `rclone --update -P -L --max-delete 0 --ignore-errors sync ...` will produce something like this:
```
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Got fatal error on delete: --max-delete threshold reached
2023-10-07 12:39:22 ERROR : Cancelling sync due to fatal error: failed to delete 24 files
2023-10-07 12:39:22 ERROR : Fatal error received - not attempting retries
Transferred:      214.195 MiB / 214.195 MiB, 100%, 898.753 KiB/s, ETA 0s
Errors:                25 (fatal error encountered)
Checks:             57320 / 57320, 100%
Deleted:                0 (files), 35 (dirs)
Transferred:           19 / 19, 100%
Elapsed time:      9m29.9s
2023/10/07 12:39:22 Failed to sync with 25 errors: last error was: failed to delete 24 files
```

I think there are 2-3 problems here, but in order of priority in my opinion:
1. Each `ERROR : Got fatal error on delete` should name the path that it had trouble deleting.
2. `2023-10-07 12:39:22 ERROR : Fatal error received - not attempting retries` leads me to believe that rclone is in fact not done, but that it stopped execution early because it encountered too many errors. I have no idea if that is the case or if it did indeed finish. Surely `--ignore-errors` should mean that it will finish.
3. I don't believe `2023-10-07 12:39:22 ERROR : Cancelling sync due to fatal error: failed to delete 24 files` is a correct statement, because I passed `--ignore-errors`?

Obviously this ticket is about point 1, but I'm not sure if the others need tickets so I won't open any.

#### How do you think rclone should be changed to solve that?
https://github.com/rclone/rclone/blob/bcb3289dad58efad617eacd2ee1103b51a6ee3cf/fs/operations/operations.go#L769
Seems to me to be the place to extend the `fs.Errorf(nil, "Got fatal error on delete: %s", err)` code to include some part of `dst`.

Thank you for reading!

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
> fs.Errorf(nil, "Got..

Changing that `nil` for `dst` will do what you want. Fancy having a go at that and sending a pull request ?

**Comment by ncw** [comment]
OK I put the file name in the error message - give this a go

[v1.65.0-beta.7420.fc231f497.fix-7355-error-delete](https://beta.rclone.org/branch/fix-7355-error-delete/v1.65.0-beta.7420.fc231f497.fix-7355-error-delete/) on branch [fix-7355-error-delete](https://github.com/rclone/rclone/tree/fix-7355-error-delete) (uploaded in 15-30 mins)

> * Each `ERROR : Got fatal error on delete` should name the path that it had trouble deleting.

Done in the above

> * `2023-10-07 12:39:22 ERROR : Fatal error received - not attempting retries` leads me to believe that rclone is in fact not done, but that it stopped execution early because it encountered too many errors. I have no idea if that is the case or if it did indeed finish. Surely `--ignore-errors` should mean that it will finish.

`--ignore-errors` means something quite specific - it doesn't mean ignore all errors everywhere!

      --ignore-errors   Delete even if there are I/O errors

So it means the sync will run the delete phase even if there were errors.

> * I don't believe `2023-10-07 12:39:22 ERROR : Cancelling sync due to fatal error: failed to delete 24 files` is a correct statement, because I passed `--ignore-errors`?

See above.
